### ISSUE TITLE
Unable to filter by node pool after update to v1.8.1

### ISSUE THREAD

**Issue by gscho** [hashicorp/nomad#23445]
### Nomad version
v1.8.1

### Issue

After updating to version 1.8.1 I can no longer filter by node pool in the nomad UI. When I select any node pool I see an error like this:

```
No jobs match your current filter selection: (NodePool == windows-amd64-instance-production). failed to create result paginator: failed to read filter expression: 1:21 (20): rule "grouping": Unmatched parentheses
```

Note this is only for node pools and all other filters I've tried seem to work fine.

### Reproduction steps

Select a node pool filter from the drop down or search for `(NodePool == my-pool)` in the search bar.

**Comment by thefallentree** [comment]
we ran into same issue, and it's because the filter code doesn't work with "-"  in the pool name,   the JS needs to quote the string

**Comment by tgross** [comment]
Thanks for the debugging there @thefallentree! I'll mark this for the attention or our UI folks.

**Comment by philrenaud** [comment]
This should be fixed by a change in #23468 — thanks for opening the issue and for your patience!
